messagebird/request_validator.py

```diff
@@ -0,0 +1,87 @@
+import hashlib
+import hmac
+import jwt
+
+from typing import Union, Dict
+from messagebird.error import ValidationError
+
+
+class RequestValidator:
+    """
+    RequestValidator validates request signature signed by MessageBird services.
+
+    See https://developers.messagebird.com/docs/verify-http-requests
+    """
+
+    ALLOWED_ALGOS = ["HS256", "HS384", "HS512"]
+
+    def __init__(self, signature_key: str, skip_url_validation: bool = False):
+        """
+        :param signature_key: customer signature key. Can be retrieved through
+        <a href="https://dashboard.messagebird.com/developers/settings">Developer Settings</a>. This is NOT your API key.
+        :param skip_url_validation: whether url_hash claim validation should be skipped.
+        Note that when true, no query parameters should be trusted.
+        """
+        super().__init__()
+        self.__signature_key = signature_key
```

Comment:
Can we use single underscores for privates? I think the rest of the lib does that too

Reply:
Thank you. Already updated.